csharp-tree-sitter.el

```diff
@@ -284,7 +286,8 @@ Key bindings:
   (tree-sitter-hl-mode))
 
 ;;;###autoload
-(add-to-list 'tree-sitter-major-mode-language-alist '(csharp-tree-sitter-mode . c-sharp))
+(with-eval-after-load 'tree-sitter
+  (add-to-list 'tree-sitter-major-mode-language-alist '(csharp-tree-sitter-mode . c-sharp)))
```

Comment:
I think you need `with-eval-after-load`  to avoid issue #208.

Reply:
Sure, we can try this.

Reply:
Do you get the error from #208? I am getting this error either from `melpa` or `quelpa`. 😕

Reply:
I don't get this error, actually. What exactly do you do to reproduce this error? I use straight, so that might be why.

Reply:
I am using `package.el` and [quelpa](), and both gives me this error. 😕 Am I the only one getting this error? But this does seem like an obvious issue so I wasn't hesitate making the bug report and PR.

Reply:
I'm having a little trouble with using `emacs -Q` on my pc right now, so I cannot verify. However, I do believe you, and if this fixes it, I say we keep it :)

Reply:
Okay, I have fully tested on my two machine. So I believed in myself and my own experience in Emacs Lisp, package.el and these stuff. Feel free to revert it if I have made a mistake here.